imbi/endpoints/project_ssm_configuration.py

```diff
@@ -0,0 +1,216 @@
+                creds['secret_access_key'], creds['session_token'])
+            for param in params:
+                if param['Name'] in params_by_path:
+                    params_by_path[
+                        param['Name']]['values'][environment] = param['Value']
+                else:
+                    params_by_path[param['Name']] = {
+                        'type': param['Type'],
+                        'values': {
+                            environment: param['Value']
+                        }
+                    }
+        if not role_arn_exists:
+            raise web.HTTPError(
+                401,
+                reason='No role ARNs exist for this namespace & environment')
+
+        output = [{
+            'name': name,
+            'type': data['type'],
+            'values': [{
+                'environment': environment,
+                'value': value
+            } for environment, value in data['values'].items()]
+        } for name, data in params_by_path.items()]
+        self.send_response(output)
+
+    async def get_aws_credentials(self,
+                                  aws_session: aioboto3.Session,
+                                  role_arn,
+                                  imbi_user: user.User,
+                                  id_token,
+                                  refresh_token,
+                                  is_retry=False) -> dict:
+        """Return (potentially cached or refreshed) AWS credentials."""
+        key = f'{imbi_user.username}:{role_arn}'
+        cached_creds = await self._redis.hgetall(key)
+        if cached_creds:
+            return {
+                key.decode('utf-8'): value.decode('utf-8')
+                for key, value in cached_creds.items()
+            }
+        try:
+            response = await aws.get_credentials(aws_session, role_arn,
+                                                 imbi_user.username, id_token)
+        except botocore.exceptions.ClientError as error:
+            if not is_retry and error.response['Error'][
+                    'Code'] == 'ExpiredTokenException':
+                id_token = await self.refresh_oauth2_tokens(
+                    imbi_user.username, imbi_user.external_id, refresh_token)
+                return await self.get_aws_credentials(aws_session, role_arn,
+                                                      imbi_user, id_token,
+                                                      refresh_token, True)
+            else:
+                error_code = error.response['Error']['Code']
+                if error_code == 'AccessDenied':
+                    raise web.HTTPError(
+                        401,
+                        reason=('Not authorized to access SSM in this '
+                                'namespace & environment'))
+                else:
+                    self.logger.error('Unexpected AWS credential failure: %s',
+                                      error_code)
```

Comment:
I would log the username and ARN here as well
```suggestion
                    self.logger.error(
                        'Unexpected AWS credential failure for %s and %s: %s',
                        imbi_user.username, role_arn, error_code)
```